### PR TITLE
Web-components: Fix custom-elements order of property application

### PR DESCRIPTION
**Pull request by sonntag-philipp**
Issues:
- https://github.com/storybookjs/storybook/issues/19167
- https://github.com/storybookjs/storybook/issues/18858

## What I did

I updated the order in which the different declaration information is extracted from the manifest. Previously, the members would override all attributes which would make attributes less specifically shown as "properties" in the storybook. This makes understanding the implementation harder.

Furthermore, I updated the `sb-button` example with a prefix slot that can be used to show the slot capabilities of the Storybook. This allows users to set a string value as content for for a slot within a component.  
To enable this functionality I had to add the `slots` declaration type to the category evaluation. In old examples, the type of the slot would default to void which would not allow the user to change it's value. This is now changed to string as default for slots, as they [don't define a type themselves in the manifest](https://github.com/webcomponents/custom-elements-manifest/blob/main/schema.d.ts#L329).

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by ndelangen** [comment]
Hello @sonntag-philipp Thank you for contributing to storybook!

I'm trying to merge in `next` into this branch but GitHub is preventing me from doing it:
```
(sonntag-philipp/issues-19167-18858-custom-elements-manifest-update) % git push                                                                                        ~/Projects/Storybook/core/code
Enumerating objects: 52, done.
Counting objects: 100% (52/52), done.
Delta compression using up to 10 threads
Compressing objects: 100% (18/18), done.
Writing objects: 100% (18/18), 2.06 KiB | 2.06 MiB/s, done.
Total 18 (delta 17), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (17/17), completed with 17 local objects.
To https://github.com/sonntag-philipp/storybook.git
 ! [remote rejected]       sonntag-philipp/issues-19167-18858-custom-elements-manifest-update -> sonntag-philipp/issues-19167-18858-custom-elements-manifest-update (refusing to allow an OAuth App to create or update workflow `.github/workflows/generate-repros-next.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/sonntag-philipp/storybook.git'
```

I don't think there's anything you or I can do to remedy that situation.. but could you merge in the `next` branch from the upstream remote? 🙏 thanks!

**Comment by sonntag-philipp** [comment]
Thank you for the feedback. I will merge next asap, but I don't think it will be within the next days.

I would really like to help, but I don't have much time right now, sorry.

**Comment by shilman** [comment]
I tried moving @sonntag-philipp 's example over to our sandboxes but it isn't working. As far as I can tell his code is working, but the example isn't. I'll see if I can pair with @kasperpeulen to figure it out.

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
@storybookjs/web-components is this good to merge?

**Comment by yannbf** [comment]
@shilman @kasperpeulen tagging you as you meant to take a look at this. I updated the branch with the latest `next`

File: code/renderers/web-components/src/docs/custom-elements.ts [hunk 1]
```diff
@@ -39,8 +39,19 @@ interface Declaration {
...
+    case 'properties':
+      type = { name: item.type?.text || item.type };
+      break;
+    case 'slots':
+      type = { name: 'string' };
+      break;
+    default:
+      type = { name: 'void' };
+      break;
+  }
```
Comment (usrrname) [inline code/renderers/web-components/src/docs/custom-elements.ts:54]: This is awesome, and sorely needed! I'd contemplate bringing this into `@storybook/html` as well

**Comment by usrrname** [comment]
WOOHOOOO

**Comment by shilman** [comment]
@usrrname better late than never 🙈

**Pull request merged**